libraries/chain/block_log.cpp

```diff
@@ -1184,46 +1186,79 @@ namespace eosio { namespace chain {
                                    } },
                        config)) {}
 
-   block_log::block_log(block_log&& other) { my = std::move(other.my); }
+   block_log::block_log(block_log&& other) noexcept { my = std::move(other.my); }
 
-   block_log::~block_log() {}
+   block_log::~block_log() = default;
 
    void     block_log::set_initial_version(uint32_t ver) { detail::block_log_impl::default_initial_version = ver; }
-   uint32_t block_log::version() const { return my->version(); }
+   uint32_t block_log::version() const {
+      std::lock_guard g(my->mtx);
+      return my->version();
+   }
 
-   void block_log::append(const signed_block_ptr& b, const block_id_type& id) { my->append(b, id, fc::raw::pack(*b)); }
+   void block_log::append(const signed_block_ptr& b, const block_id_type& id) {
+      std::vector<char> packed_block = fc::raw::pack(*b);
```

Comment:
Should this line be protected too? Would someone modify `b` while `pack` is running?

Reply:
We would have to protect all access to signed_block to guard against that. We already assume that signed_block is not modified after finalization. We already use signed_block on other threads after finalization.